src/languages/db2i/db2i.formatter.ts

```diff
@@ -0,0 +1,193 @@
+import { functions } from './db2i.functions.js';
+import { keywords } from './db2i.keywords.js';
+
+const reservedSelect = expandPhrases(['SELECT [ALL | DISTINCT]']);
+
+const reservedClauses = expandPhrases([
+  // queries
+  'WITH',
+  'FROM',
+  'WHERE',
+  'GROUP BY',
+  'HAVING',
+  'PARTITION BY',
+  'ORDER [SIBLINGS] BY [INPUT SEQUENCE]',
+  'OFFSET {ROW | ROWS}',
+  'FETCH {FIRST | NEXT} {ROW | ROWS} ONLY',
+  'LIMIT',
+  'FOR UPDATE [OF]',
+  'FOR READ ONLY',
+  'OPTIMIZE FOR [ALL] {ROW | ROWS}',
+  // Data modification
+  // - insert:
+  'INSERT INTO',
+  'VALUES',
+  // - update:
+  'SET',
+  // - merge:
+  'MERGE INTO',
+  'WHEN [NOT] MATCHED [THEN]',
+  'UPDATE SET',
+  'DELETE',
+  'INSERT',
+  // Data definition
+  'CREATE [OR REPLACE] TABLE [FOR SYSTEM NAME]',
+  'CREATE [OR REPLACE] [RECURSIVE] VIEW [FOR SYSTEM NAME]',
+]);
+
+const onelineClauses = expandPhrases([
+  // - update:
+  'UPDATE',
+  'WHERE CURRENT OF',
+  'WITH {NC | RR | RS | CS | UR}',
+  // - delete:
+  'DELETE FROM',
+  // - drop table:
+  'DROP TABLE',
+  // alter table:
+  'ALTER TABLE',
+  'ADD [COLUMN]',
+  'ALTER [COLUMN]',
+  'DROP [COLUMN]',
+  'RENAME [COLUMN]',
+  'SET DATA TYPE', // for alter column
+  'SET {GENERATED ALWAYS | GENERATED BY DEFAULT}', // for alter column
+  'SET NOT NULL', // for alter column
+  'SET {NOT HIDDEN | IMPLICITLY HIDDEN}', // for alter column
+  'SET FIELDPROC', // for alter column
+  'DROP {DEFAULT | NOT NULL | GENERATED | IDENTITY | ROW CHANGE TIMESTAMP | FIELDPROC}', // for alter column
+  // - truncate:
+  'TRUNCATE [TABLE]',
+  // other
+  'SET [CURRENT] SCHEMA',
+  // https://www.ibm.com/docs/en/i/7.5?topic=reference-statements
+  'ALLOCATE CURSOR',
+  'ALLOCATE DESCRIPTOR',
+  'ALTER FUNCTION',
+  'ALTER MASK',
+  'ALTER PERMISSION',
+  'ALTER PROCEDURE',
+  'ALTER SEQUENCE',
+  'ALTER TRIGGER',
+  'ASSOCIATE LOCATORS',
+  'BEGIN DECLARE SECTION',
+  'CALL',
+  'CLOSE',
+  'COMMENT',
+  'COMMIT',
+  'CONNECT',
+  'CREATE [OR REPLACE] ALIAS',
+  'CREATE [OR REPLACE] FUNCTION',
+  'CREATE INDEX',
+  'CREATE [OR REPLACE] MASK',
+  'CREATE [OR REPLACE] PERMISSION',
+  'CREATE [OR REPLACE] PROCEDURE',
+  'CREATE SCHEMA',
+  'CREATE [OR REPLACE] SEQUENCE',
+  'CREATE [OR REPLACE] TRIGGER',
+  'CREATE TYPE',
+  'CREATE [OR REPLACE] VARIABLE',
+  'DEALLOCATE DESCRIPTOR',
+  'DECLARE CURSOR',
+  'DECLARE GLOBAL TEMPORARY TABLE',
+  'DECLARE PROCEDURE',
+  'DECLARE STATEMENT',
+  'DECLARE VARIABLE',
+  'DESCRIBE',
+  'DESCRIBE CURSOR',
+  'DESCRIBE INPUT',
+  'DESCRIBE OUTPUT',
+  'DESCRIBE PROCEDURE',
+  'DESCRIBE TABLE',
+  'DISCONNECT',
+  'DROP',
+  'END DECLARE SECTION',
+  'EXECUTE',
+  'EXECUTE IMMEDIATE',
+  'FETCH',
+  'FREE LOCATOR',
+  'GET DESCRIPTOR',
+  'GET DIAGNOSTICS',
+  'GRANT',
+  'HOLD LOCATOR',
+  'INCLUDE',
+  'INSERT',
+  'LABEL',
+  'LOCK TABLE',
+  'OPEN',
+  'PREPARE',
+  'REFRESH TABLE',
+  'RELEASE',
+  'RELEASE SAVEPOINT',
+  'RENAME',
+  'REVOKE',
+  'ROLLBACK',
+  'SAVEPOINT',
+  'SELECT',
```

Comment:
The problem is that `SELECT` is listed both here in `onelineClauses` and also above in `selectClause`.

The entry in here happens to have higher precedence and the result is that column names are listed one the same line as `SELECT` keyword.

Reply:
Ah, this I forgot to remove from the list, when adding the statements. Fixed...